### PR TITLE
 Remove virtualenv and pip from requirements

### PR DESCRIPTION
**Pull request by paulcartier**
- Install virtualenv during installation step
- Update requirements in the documentation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,9 +5,10 @@ It uses [Hammr CLI](https://github.com/usharesoft/hammr) to create, generate and
 
 ## Prerequisite
 - The plugin requires at least Jenkins version 2.150.2 to work.
-- `pip`,`virtualenv` and `python2.7` need to be installed on the executor.
+- `python2.7` needs to be installed on the executor.
 - You also need a running UForge AppCenter server (minimum version: 3.8.13).
 
+If the "Installation Step" of the plugin fails, you may need to install some more dependencies on your Jenkins executor. See: http://docs.usharesoft.com/projects/hammr/en/latest/pages/getting-started/installation.html#installing-hammr for more information.
```
Comment (lqueiroga) [inline README.md]: I know I suggested this sentence but I'm wondering if it's a good idea to link hammr documentation with "install python" step...

File: src/main/java/com/usharesoft/jenkins/steps/InstallStep.java [hunk 2]
```diff
@@ -18,22 +18,43 @@ public InstallStep(UForgeLauncher launcher, String version) {
...
         launcher.launchInstall(getInstallHammrCmd(), true);
     }
 
-    ArgumentListBuilder getInstallVenvCmd() {
+    ArgumentListBuilder getDownloadVenvCmd() {
         ArgumentListBuilder args = new ArgumentListBuilder();
-        args.add("virtualenv");
-        args.add("--python=python2.7");
-        args.add(launcher.getScriptWorkspace());
+        args.add("curl");
```
Comment (lqueiroga) [inline src/main/java/com/usharesoft/jenkins/steps/InstallStep.java]: Are we sure that curl is installed?

File: src/main/java/com/usharesoft/jenkins/steps/InstallStep.java [hunk 3]
```diff
@@ -18,22 +18,43 @@ public InstallStep(UForgeLauncher launcher, String version) {
...
+        args.add("--location");
+        args.add("--output").add("virtualenv.tar.gz");
+        args.add("https://github.com/pypa/virtualenv/tarball/12.0.7");
```
Comment (lqueiroga) [inline src/main/java/com/usharesoft/jenkins/steps/InstallStep.java]: If we decide to fix the venv version here, why we don't get the latest one today: 16.6.1

File: src/main/java/com/usharesoft/jenkins/steps/InstallStep.java [hunk 4]
```diff
@@ -18,22 +18,43 @@ public InstallStep(UForgeLauncher launcher, String version) {
...
+        args.add("xvfz");
+        args.add("virtualenv.tar.gz");
+
+        return args;
+    }
+
+    ArgumentListBuilder getInitVenvCmd() {
+        ArgumentListBuilder args = new ArgumentListBuilder();
+        args.add("python2.7");
+        args.add("pypa-virtualenv-5921a37/virtualenv.py");
```
Comment (lqueiroga) [inline src/main/java/com/usharesoft/jenkins/steps/InstallStep.java]: Is it a random name?

**Review by lqueiroga — CHANGES_REQUESTED** [review]

**Review by alefebvr — DISMISSED** [review]
Looks good to me.

File: src/main/java/com/usharesoft/jenkins/steps/InstallStep.java (hunk 3 above)
Comment (paulcartier) [reply src/main/java/com/usharesoft/jenkins/steps/InstallStep.java]: True, I'll make the change

File: src/main/java/com/usharesoft/jenkins/steps/InstallStep.java (hunk 4 above)
Comment (paulcartier) [reply src/main/java/com/usharesoft/jenkins/steps/InstallStep.java]: No, it is the name of the directory located in the archive. It is always the same so I hardcoded it.

File: src/main/java/com/usharesoft/jenkins/steps/InstallStep.java [hunk 5]
```diff
@@ -18,22 +22,45 @@ public InstallStep(UForgeLauncher launcher, String version) {
...
     }
 
-    ArgumentListBuilder getInstallVenvCmd() {
+    void downloadVirtualenv() {
+        try {
+            FileUtils.copyURLToFile(
+                    new URL("https://github.com/pypa/virtualenv/tarball/16.6.1"),
+                    new File(launcher.getWorkspace() + "/virtualenv.tar.gz"),
+                    1000,
+                    1000);
```
Comment (lqueiroga) [inline src/main/java/com/usharesoft/jenkins/steps/InstallStep.java]: only 1 second? Seems to short for me.

File: src/main/java/com/usharesoft/jenkins/steps/InstallStep.java [hunk 6]
```diff
@@ -18,22 +22,45 @@ public InstallStep(UForgeLauncher launcher, String version) {
...
+            e.printStackTrace();
+        }
+    }
+
+    ArgumentListBuilder getExtractVenvCmd() {
+        ArgumentListBuilder args = new ArgumentListBuilder();
+        args.add("tar");
+        args.add("xvfz");
+        args.add("virtualenv.tar.gz");
+
```
Comment (lqueiroga) [inline src/main/java/com/usharesoft/jenkins/steps/InstallStep.java]: You can untar with java code too, with apache commons-compress

**Review by lqueiroga — CHANGES_REQUESTED** [review]

**Review by lqueiroga — APPROVED** [review]

**Comment by lqueiroga** [comment]
Thanks for this PR

**Pull request merged**